## Taylor Swift Spotify Dataset - data preprocessing
The data used comes from [Kaggle](https://www.kaggle.com/datasets/jarredpriester/taylor-swift-spotify-dataset). This is a 
passion project used to learn the workflow and the rules of a proper analysis.

In [31]:
import pandas as pd
import numpy as np

songs = pd.read_csv('../data/taylor_swift_spotify.csv')
print("Shape of the data:")
print(songs.shape)
print()

print("Columns:")
print(songs.columns)

Shape of the data:
(582, 18)

Columns:
Index(['Unnamed: 0', 'name', 'album', 'release_date', 'track_number', 'id',
       'uri', 'acousticness', 'danceability', 'energy', 'instrumentalness',
       'liveness', 'loudness', 'speechiness', 'tempo', 'valence', 'popularity',
       'duration_ms'],
      dtype='object')


The songs count is 582 with 18 columns describing them. The final goal of this project is to create an AI model
that correctly predicts in which era the song was released, therefore we remove columns that are not numerical,
such as realease_date or track_number. 

In [32]:
songs_info = songs.drop(['Unnamed: 0', 'release_date', 'track_number', 'id', 'uri'], axis=1)

Due to deluxe albums having extra content voice memos are present in the dataset. They are early versions
of songs that already exist in the dataset and have 0 popularity, so they do not serve any purpose.

In [33]:
print(songs_info[songs_info['popularity'] == 0]['name'])
print()

songs_info = songs_info[songs_info['popularity'] != 0]
print("Shape after deleting the irrelevant columns and voice memos:")
print(songs_info.shape)

418       I Know Places - Voice Memo
419    I Wish You Would - Voice Memo
420         Blank Space - Voice Memo
Name: name, dtype: object

Shape after deleting the irrelevant columns and voice memos
(579, 13)


In [ ]:
album_full_names = ['THE TORTURED POETS DEPARTMENT: THE ANTHOLOGY', 
                    'Midnights (3am Edition)', 
                    'evermore (deluxe version)', 
                    'folklore (deluxe version)',
                    'Lover', 'reputation', 
                    '1989 (Deluxe)', 
                    'Red', 
                    'Speak Now', 
                    'Fearless (Platinum Edition)', 
                    'Taylor Swift (Deluxe Edition)']

songs_info = songs_info[songs_info['album'].isin(album_full_names)]

In [ ]:
album_short_names = ['ttpd', 'midnights', 'evermore', 'folklore', 'lover', 'rep', '1989', 'red', 'speak now', 'fearless', 'debut']
album_dict = dict(zip(album_full_names, album_short_names))
songs_info['album'] = songs_info['album'].replace(album_dict)
songs_info.to_csv('../data/taylor_songs_no_TV.csv', index=False)